lars-reimann/api-editor
# feat: pass path of API sources to `api` command

Closes #524.

### Summary of Changes

* Stricter config for mypy (type hints were missing).
* Optionally pass path to API source to `api` and `all` commands.
* Remove `scikit-learn` dependency. To generate data for `scikit-learn` now, we can download the sources from [GitHub](https://github.com/scikit-learn/scikit-learn), for example.
* Use our own code as input for smoke tests, which makes them considerably faster than using `scikit-learn`.

### Testing Instructions

1. Open the correct directory in a terminal:
```sh
cd package-parser
```
2. Run the `api` command:
```sh
poetry run parse-package api -p package_parser -s package_parser -o out
```
3. Run the `all` command:
```sh
poetry run parse-package all -p package_parser -s package_parser -c package_parser -o out
```
4. Check that everything ran correctly.
